Unused variable warnings

```
/Users/selsta/dev/monero/src/crypto/hash-extra-jh.c:40:7: warning: unused variable 'r' [-Wunused-variable]
  int r = jh_hash(HASH_SIZE * 8, data, 8 * length, (uint8_t*)hash);
      ^
1 warning generated.
```

```
/Users/selsta/dev/monero/src/crypto/hash-extra-skein.c:38:7: warning: unused variable 'r' [-Wunused-variable]
  int r = skein_hash(8 * HASH_SIZE, data, 8 * length, (uint8_t*)hash);
      ^
1 warning generated.
```

```
/Users/selsta/dev/monero/external/supercop/crypto_sign/ed25519/amd64-64-24k/ge25519_scalarmult_base.c:15:22: warning: unused variable 'ecd' [-Wunused-const-variable]
static const fe25519 ecd = {{0x75EB4DCA135978A3, 0x00700A4D4141D8AB, 0x8CC740797779E898, 0x52036CEE2B6FFE73}};
                     ^
1 warning generated.
```

I have been fixing warnings for the monero repo and these show up on Mac clang.

--- a/crypto_sign/ed25519/amd64-64-24k/ge25519_scalarmult_base.c
+++ b/crypto_sign/ed25519/amd64-64-24k/ge25519_scalarmult_base.c
@@ -12,7 +12,9 @@ static const ge25519_niels ge25519_base_multiples_niels[] = {
 };
 
 /* d */
+#ifdef SMALLTABLES
 static const fe25519 ecd = {{0x75EB4DCA135978A3, 0x00700A4D4141D8AB, 0x8CC740797779E898, 0x52036CEE2B6FFE73}};
+#endif
 
 void ge25519_scalarmult_base(ge25519_p3 *r, const sc25519 *s)
 {

--- a/crypto_sign/ed25519/amd64-64-24k/sc25519.h
+++ b/crypto_sign/ed25519/amd64-64-24k/sc25519.h
@@ -55,7 +55,7 @@ void sc25519_mul_shortsc(sc25519 *r, const sc25519 *x, const shortsc25519 *y);
 /* Convert s into a representation of the form \sum_{i=0}^{63}r[i]2^(4*i)
  * with r[i] in {-8,...,7}
  */
-void sc25519_window4(signed char r[85], const sc25519 *s);
+void sc25519_window4(signed char r[64], const sc25519 *s);
 
 void sc25519_slide(signed char r[256], const sc25519 *s, int swindowsize);
 

